lora-phy/src/sx127x/sx1276.rs

```diff
@@ -108,7 +129,48 @@ impl Sx127xVariant for Sx1276 {
         }
         let mut config_3 = radio.read_register(Register::RegModemConfig3).await?;
         config_3 = (config_3 & 0xf3u8) | ldro_agc_auto_flags;
-        radio.write_register(Register::RegModemConfig3, config_3).await
+        radio.write_register(Register::RegModemConfig3, config_3).await?;
+
+        if radio.data.sensitivity_quirk {
+            // apply Errata 2.1: Sensitivity optimization with 500 kHz bandwidth
+
+            if mdltn_params.bandwidth == Bandwidth::_500KHz {
+                let val_1 = 0x02;
+                match mdltn_params.frequency_in_hz {
+                    862..=1020 => {
+                        radio
+                            .write_register(Register::RegHighBwOptimize1, val_1)
+                            .await?;
+
+                        radio
+                            .write_register(Register::RegHighBwOptimize2, 0x64)
+                            .await?;
+                        return Ok(())
+                    },
+                    410..=525 => {
+                        radio
+                            .write_register(Register::RegHighBwOptimize1, val_1)
+                            .await?;
+                        radio
+                            .write_register(Register::RegHighBwOptimize2, 0x7f)
+                            .await?;
+                        return Ok(())
+                    }
+                    // fall through...
+                    _ => {}
+                }
+            }
+
+            // for all other combinations of bandwidth / frequencies, reset to 0x03 and the value at
```

Comment:
Please clean up the comment.

And `cargo fmt` should clean up the extra newlines.